pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -283,6 +286,16 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		qosContainerManager: qosContainerManager,
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.TopologyManager) {
+		klog.Infof("Node Config: %v", nodeConfig)
+		cm.topologyManager = topologymanager.NewManager(
+			nodeConfig.ExperimentalTopologyManagerPolicy,
+		)
+		klog.Infof("[topologymanager] Initilizing Topology Manager with %s policy", nodeConfig.ExperimentalTopologyManagerPolicy)
+	} else {
+		cm.topologyManager = topologymanager.NewFakeManager()
```

Comment:
In the other OSes, we return a nil `topologyManager`.  I don't think we want to use a fake manager outside of test code right?

Reply:
+1

Reply:
From what I can see, the other OSes don't have TopologyManager is part of the containerManagerImpl so there is no reference, so we don't return anything?

 I was following the convention used by DeviceManager which returns a ManagerStub below. There is also a reference to the Fake CPU Manager in the other OSes when returning the internalContainerLifecycleImpl.

If I return nil, the code errors as cm.TopologyManager functions are called below. So I could wrap those calls in a check for the feature gate but looks like the fake/stub managers are used outside of test code.

Reply:
I actually think the only call you would need to wrap is this one (which probably makes sense to wrap anyway):

```
pkg/kubelet/kubelet.go:
klet.admitHandlers.AddPodAdmitHandler(klet.containerManager.GetTopologyPodAdmitHandler())
```

Looking through the code, I don't see anywhere else that `cm.topologyManager` is referenced (other than https://github.com/kubernetes/kubernetes/pull/74357/files#diff-06aa1b61a9bc4a50aa8d8b0a69d95bb2R347), whose internal references to the `topologyManager` are already protected by a check on the feature gate:

https://github.com/kubernetes/kubernetes/pull/74357/files#diff-81cef540f416b0451c18470420adc42dR40

Reply:
Ah yes I see the code I was referring to is actually in the CPU and Device Manager PRs.
https://github.com/kubernetes/kubernetes/pull/73920 & https://github.com/kubernetes/kubernetes/pull/74345 have the references to topology manager to register them as Hint Providers

Reply:
It still feels more natural to me to remove the `FakeManager()` here and protect all of the accesses to `cm.topologymanager` with a check on the feature gate, but I will leave it to @sjennings for the final call.

Reply:
we can clean it up in a follow on.  there are a number of things that can be cleaned up.  i'm making a list :)